src/main/java/org/opensearch/ad/ExecuteADResultResponseRecorder.java

```diff
@@ -0,0 +1,268 @@
+            User user = detector.getUser();
+
+            if (response.getError() != null) {
+                log.info("Anomaly result action run successfully for {} with error {}", detectorId, response.getError());
+            }
+
+            AnomalyResult anomalyResult = response
+                .toAnomalyResult(
+                    detectorId,
+                    dataStartTime,
+                    dataEndTime,
+                    executionStartTime,
+                    Instant.now(),
+                    anomalyDetectionIndices.getSchemaVersion(ADIndex.RESULT),
+                    user,
+                    response.getError()
+                );
+
+            String resultIndex = detector.getResultIndex();
+            anomalyResultHandler.index(anomalyResult, detectorId, resultIndex);
+            updateRealtimeTask(response, detectorId);
+        } catch (EndRunException e) {
+            throw e;
+        } catch (Exception e) {
+            log.error("Failed to index anomaly result for " + detectorId, e);
+        }
+    }
+
+    private void updateRealtimeTask(AnomalyResultResponse response, String detectorId) {
+        if (response.isHCDetector() != null && response.isHCDetector()) {
+            if (adTaskManager.skipUpdateHCRealtimeTask(detectorId, response.getError())) {
+                return;
+            }
+            DiscoveryNode[] dataNodes = nodeFilter.getEligibleDataNodes();
+            Set<DetectorProfileName> profiles = new HashSet<>();
+            profiles.add(DetectorProfileName.INIT_PROGRESS);
+            ProfileRequest profileRequest = new ProfileRequest(detectorId, profiles, true, dataNodes);
+            Runnable profileHCInitProgress = () -> {
+                client.execute(ProfileAction.INSTANCE, profileRequest, ActionListener.wrap(r -> {
+                    log.debug("Update latest realtime task for HC detector {}, total updates: {}", detectorId, r.getTotalUpdates());
+                    updateLatestRealtimeTask(
+                        detectorId,
+                        null,
+                        r.getTotalUpdates(),
+                        response.getDetectorIntervalInMinutes(),
+                        response.getError()
+                    );
+                }, e -> { log.error("Failed to update latest realtime task for " + detectorId, e); }));
+            };
+            if (!adTaskManager.isHCRealtimeTaskStartInitializing(detectorId)) {
+                // real time init progress is 0 may mean this is a newly started detector
+                // Delay real time cache update by one minute. If we are in init status, the delay may give the model training time to
+                // finish. We can change the detector running immediately instead of waiting for the next interval.
+                threadPool.schedule(profileHCInitProgress, new TimeValue(60, TimeUnit.SECONDS), AnomalyDetectorPlugin.AD_THREAD_POOL_NAME);
```

Comment:
Previous PR had this extra 60 time wait when `response.getAnomalyScore() <= 0 || Double.isNaN(response.getAnomalyScore())) && response.getError() == null` was true. 
1. Is limiting this wait time to when init is 0 for hcad and when single stream throws `ResourceNotFoundException` narrowing the scope of this wait only happening when detector is first being created? 
2. Just to make sure we aren't going to get the discrepancy that sometimes happen between profile api and real time task api cause here we are internally checking if the realtime task already exists and we are above init of 0?

Reply:
1. yes, you are right.
2. I haven't fixed the discrepancy issue. Tracked here: https://github.com/opensearch-project/anomaly-detection/issues/502 Plan to do it after the speed up PR is merged so the PR is not too big.